### ISSUE TITLE
opening_hours.intervals(start, end) does not include intervals after last midnight in the specified range

### ISSUE THREAD

**Issue by HaramanJohal** [remi-dupre/opening-hours-rs#52]
Hello `opening-hours-rs` 👋🏾 
I found an issue where `opening_hours.intervals(start, end)` does not include intervals after last midnight between `start` and `end`.

Current behaviour is:
```
>>> list(OpeningHours('Mo-Su 00:00-06:00, 23:00-00:00').intervals(datetime(2024, 11, 11, 1, 0),datetime(2024, 11, 12, 1, 0)))
[
  (datetime.datetime(2024, 11, 11, 1, 0), datetime.datetime(2024, 11, 11, 6, 0), 'open', []),
  (datetime.datetime(2024, 11, 11, 6, 0), datetime.datetime(2024, 11, 11, 23, 0), 'closed', []),
  (datetime.datetime(2024, 11, 11, 23, 0), datetime.datetime(2024, 11, 12, 0, 0), 'open', [])
]
```
We would expect to see a fourth interval:
```(datetime.datetime(2024, 11, 12, 0, 0), datetime.datetime(2024, 11, 12, 1, 0), 'open', [])```
or maybe a longer third interval:
```(datetime.datetime(2024, 11, 11, 23, 0), datetime.datetime(2024, 11, 12, 1, 0), 'open', [])```

More broadly `intervals` seems to run up to the last midnight included between `start` and `end`.
This example works fine because end is at a midnight:
```
>>> list(OpeningHours('Mo-Su 23:00-06:00').intervals(datetime(2024, 11, 11, 1, 0),datetime(2024, 11, 13, 0, 0)))
[
  (datetime.datetime(2024, 11, 11, 1, 0), datetime.datetime(2024, 11, 11, 6, 0), 'open', []),
  (datetime.datetime(2024, 11, 11, 6, 0), datetime.datetime(2024, 11, 11, 23, 0), 'closed', []),
  (datetime.datetime(2024, 11, 11, 23, 0), datetime.datetime(2024, 11, 12, 6, 0), 'open', []),
  (datetime.datetime(2024, 11, 12, 6, 0), datetime.datetime(2024, 11, 12, 23, 0), 'closed', []),
  (datetime.datetime(2024, 11, 12, 23, 0), datetime.datetime(2024, 11, 13, 0, 0), 'open', [])
]
```
This example again misses out the final interval because `end` runs into the next day:
```
>>> list(OpeningHours('Mo-Su 23:00-06:00').intervals(datetime(2024, 11, 11, 1, 0),datetime(2024, 11, 13, 5, 0)))
[
  (datetime.datetime(2024, 11, 11, 1, 0), datetime.datetime(2024, 11, 11, 6, 0), 'open', []),
  (datetime.datetime(2024, 11, 11, 6, 0), datetime.datetime(2024, 11, 11, 23, 0), 'closed', []),
  (datetime.datetime(2024, 11, 11, 23, 0), datetime.datetime(2024, 11, 12, 6, 0), 'open', []),
  (datetime.datetime(2024, 11, 12, 6, 0), datetime.datetime(2024, 11, 12, 23, 0), 'closed', []),
  (datetime.datetime(2024, 11, 12, 23, 0), datetime.datetime(2024, 11, 13, 0, 0), 'open', [])
]
```

(Note for the above examples I've also tested with a couple different tariff formats like 'Mo-Su 00:00-06:00, 23:00-24:00' or 'Mo-Su 23:00-06:00' and got the same results)

It would be great to fix this so that `intervals` can return correct values
Thanks for your time

**Comment by remi-dupre** [comment]
Hi!

Thanks for reporting the issue with such precision.

This should be fixed with version 0.8.3 which will be published in a few minutes :+1:

**Comment by HaramanJohal** [comment]
Thanks for the speedy response!
